tool/live-monitor.js

```diff
@@ -0,0 +1,172 @@
+        } else {
+            throw new Error(`Parameter '${param}' not recognized`);
+        }
+        return params;
+    }, defaultParamsValues);
+    return params;
+};
+
+const PEGOUT_METHOD_SIGNATURES = {
+    releaseBtc: '0x',
+    updateCollections: '0x0c5a9990',
+    addSignature: '0xf10b9c59'
+};
+
+const PEGIN_METHOD_SIGNATURES = {
+    registerBtcCoinbaseTransaction: '0xccf417ae',
+    registerBtcTransaction: '0x43dc0656',
+    registerFastBridgeBtcTransaction: '0x6adc0133'
+};
+
+function isAPegoutRelatedTransactionData(data) {
+    const methodSignature = data.slice(0, 10);
+    switch(methodSignature) {
+        case PEGOUT_METHOD_SIGNATURES.releaseBtc:
+        case PEGOUT_METHOD_SIGNATURES.updateCollections:
+        case PEGOUT_METHOD_SIGNATURES.addSignature:
+            return true;
+        default:
+            return false;
+    }
+}
+
+function isAPeginRelatedTransactionData(data) {
+    const methodSignature = data.slice(0, 10);
+    switch(methodSignature) {
+        case PEGIN_METHOD_SIGNATURES.registerBtcTransaction:
+        case PEGIN_METHOD_SIGNATURES.registerBtcCoinbaseTransaction:
+            return true;
+        default:
+            return false;
+    }
+}
+
+const params = getParsedParams();
+
+const web3Client = new Web3(networkParser(params.network));
+
+let currentBlockNumber = params.fromBlock;
+
+let notified = false;
+
+async function monitor() {
+
+    try {
+
+        const latestBlockNumber = await web3Client.eth.getBlockNumber();
+
+        if(currentBlockNumber === 'latest') {
+            currentBlockNumber = latestBlockNumber;
+        }
+
+        if(latestBlockNumber < currentBlockNumber) {
+            if(!notified) {
+                console.warn("latestBlockNumber was reached!");
+                notified = true;
+            }
+            return;
+        }
+
+        notified = false;
+
+        const block = await web3Client.eth.getBlock(currentBlockNumber, true);
+
+        console.info("Checking block: ", block.number);
+
+        currentBlockNumber++;
+
+        for(const transaction of block.transactions) {
+            
+            if(transaction.to === Bridge.address) {
+
+                const isPeginRelated = isAPeginRelatedTransactionData(transaction.input);
+                const isPegoutRelated = isAPegoutRelatedTransactionData(transaction.input);
+
+                // Requested only pegouts, if tx is not a pegout then continue
+                if(!params.pegin && (params.pegout && !isPegoutRelated)) {
+                    continue;
+                // Requested only pegins, if tx is not a pegin then return
+                } else if(!params.pegout && (params.pegin && !isPeginRelated)) {
+                    continue;
+                // Requested only pegins and pegouts, if tx is not a pegin or pegout then return
+                } else if((params.pegin && !isPeginRelated) || (params.pegout && !isPegoutRelated)) {
+                    continue;
+                }
```

Comment:
```
if((params.pegin && !isPeginRelated) || (params.pegout && !isPegoutRelated)) {
                continue;
            }
```
Doesn't that cover it? Or am I missing something?

Reply:
That's the same as the last `else if`. I don't think it would make a difference.

Reply:
I think Marcos it's right, the last if `else if((params.pegin && !isPeginRelated) || (params.pegout && !isPegoutRelated))` cover the two if above, so it's enough only with that.